kyo-offheap/native/src/main/java/lang/foreign/MemorySegment.scala

```diff
@@ -0,0 +1,142 @@
+        require(offset + unsafe.sizeOf[CShort] <= byteSize)
+        !(ptr + offset).cast[Short] = value
+    def set(layout: ValueLayout.ofInt, offset: Long, value: Int): Unit =
+        require(offset + unsafe.sizeOf[CInt] <= byteSize)
+        !(ptr + offset).cast[Int] = value
+    def set(layout: ValueLayout.ofLong, offset: Long, value: Long): Unit =
+        require(offset + unsafe.sizeOf[CLong] <= byteSize)
+        !(ptr + offset).cast[Long] = value
+    def set(layout: ValueLayout.ofFloat, offset: Long, value: Float): Unit =
+        require(offset + unsafe.sizeOf[CFloat] <= byteSize)
+        !(ptr + offset).cast[Float] = value
+    def set(layout: ValueLayout.ofDouble, offset: Long, value: Double): Unit =
+        require(offset + unsafe.sizeOf[CDouble] <= byteSize)
+        !(ptr + offset).cast[Double] = value
+    def set(layout: ValueLayout.ofChar, offset: Long, value: Char): Unit =
+        require(offset + unsafe.sizeOf[CChar] <= byteSize)
+        !(ptr + offset).cast[Char] = value
+    def set(layout: AddressLayout, offset: Long, value: MemorySegment): Unit =
+        !(ptr + offset).cast[Ptr[Byte]] = value.ptr
+end MemorySegment
+
+object MemorySegment:
+    private[foreign] def allocate(byteSize: Long): MemorySegment =
+        val ptr = malloc(byteSize).asInstanceOf[Ptr[Byte]]
+        if ptr == null then throw new RuntimeException("malloc returned null")
+        new MemorySegment(ptr, byteSize)
+    end allocate
+
+    /** Copies a block of memory from the source segment to the destination segment.
+      *
+      * @param srcSegment
+      *   The source MemorySegment.
+      * @param srcOffset
+      *   The offset (in bytes) into the source segment.
+      * @param dstSegment
+      *   The destination MemorySegment.
+      * @param dstOffset
+      *   The offset (in bytes) into the destination segment.
+      * @param bytes
+      *   The number of bytes to copy.
+      */
+    def copy(srcSegment: MemorySegment, srcOffset: Long, dstSegment: MemorySegment, dstOffset: Long, bytes: Long): Unit =
+        memcpy(dstSegment.ptr + dstOffset, srcSegment.ptr + srcOffset, bytes.toULong)
+        ()
+end MemorySegment
+
+/** Minimal inline stub for ValueLayout. This defines constants for the primitive types.
+  */
+sealed trait ValueLayout
+
+object ValueLayout:
+    sealed trait ofBoolean extends ValueLayout
+    sealed trait ofByte    extends ValueLayout
+    sealed trait ofChar    extends ValueLayout
+    sealed trait ofShort   extends ValueLayout
+    sealed trait ofInt     extends ValueLayout
+    sealed trait ofLong    extends ValueLayout
+    sealed trait ofFloat   extends ValueLayout
+    sealed trait ofDouble  extends ValueLayout
+
+    case object JAVA_BOOLEAN          extends ValueLayout
+    case object JAVA_BYTE             extends ValueLayout
+    case object JAVA_CHAR             extends ValueLayout
+    case object JAVA_CHAR_UNALIGNED   extends ValueLayout
```

Comment:
When I think of it again, if we're not going to implement the behavior for them, we should probably not define these `_UNALIGNED` objects to begin with.
```suggestion
```